---
title: no-woof
description: No woof!
---

<script setup lang="ts">
import CodeEditor from '../../.vitepress/theme/components/code-editor.vue';
import {ruleName, presetConfigs, initialText} from '../../src/sample-code/no-woof.js';
</script>

> "Woof" â€” Randalf T.

# Disallow woof! (`no-woof`)

ğŸ’¼ This rule is enabled in the following [configs](/configs/): ğŸŒ `all`, âœ… `recommended`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## ğŸ“– Rule details

No woof!

## ğŸ’¡ Examples

```js
// âŒ Incorrect
// woof Woof! WOOF!
const woof = 'woof'

// âœ… Correct
const foo = 'bar'
```

## ğŸ”§ Config

```js
{ rules: { 'ninja/no-woof': 2 } }
```

## ğŸ”— See also

- [private joke](https://en.wikipedia.org/wiki/Private_joke)

## ğŸ§‘â€ğŸ’» Demo

<CodeEditor :rule="ruleName" :text="initialText" :presetConfigs="presetConfigs" />
