---
title: optimize-string-ternary
description: Disallow repetition when building strings with ternaries=
---

<script setup lang="ts">
import CodeEditor from '../../.vitepress/theme/components/code-editor.vue';
import {ruleName, presetConfigs, initialText} from '../../src/sample-code/optimize-string-ternary';
</script>

> Duplication is the primary enemy of a well-designed system. â€” Robert C. Martin

# Disallow repetition when building strings with ternaries (`optimize-string-ternary`)

ğŸ’¼ This rule is enabled in the following [configs](/configs/): ğŸŒ `all`, âœ…
`recommended`.

ğŸ”§ This rule is automatically fixable by the
[`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## ğŸ“– Rule details

This rule disallows repetition when building strings with ternaries.

## ğŸ’¡ Examples

```js
// âŒ Incorrect
const CauseOrCampaign = !isNotCause ? 'Cause' : 'Campaign'

// âœ… Correct
const CauseOrCampaign = 'Ca' + (!isNotCause ? 'use' : 'mpaign')
```

## ğŸ”§ Config

```js
{ rules: { 'ninja/optimize-string-ternary': 2 } }
```

## ğŸ§‘â€ğŸ’» Demo

<CodeEditor :rule="ruleName" :text="initialText" :presetConfigs="presetConfigs" />
