---
title: tab
description: Use tabs to separate words in the code
---

<script setup lang="ts">
import CodeEditor from '../../.vitepress/theme/components/code-editor.vue';
import {ruleName, presetConfigs, initialText} from '../../src/sample-code/tab.js';
</script>

> "I just think it's weird that you use spaces instead of tabs." â€” Richard Hendricks

# Require word separators to be tabs, not spaces (`dont/tab`)

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## ğŸ“– Rule details

Tabs are better than spaces. However and unfortunately, most developers still use spaces to separate words.

## ğŸ’¡ Examples

```js
// âŒ Incorrect
if (a) {
  b = c;
  function foo(d) {
    e = f;
  }
}

// âœ… Correct
if	(a)	{
		b	=	c;
		function	foo(d)	{
				e	=	f;
		}
}
```

## ğŸ”§ Config

```js
{ rules: { 'dont/tab': 2 } }
```

## ğŸ§‘â€ğŸ’» Demo

<CodeEditor :rule="ruleName" :text="initialText" :presetConfigs="presetConfigs" />
